File: RandomTest.c
#include <stdio.h>
#include <sys/stat.h>
#include <fcntl.h>
#include <string.h>
#include <errno.h>
#include <stdlib.h>
#include <time.h>
#include <unistd.h>
extern int errno;
char* convert_decimal_binary(int a){
    if(a<0){
        a =  (a+128) + 128;
    }
    char* result = (char*)(malloc(sizeof(char)*9)); 
    for(int i=0;i<8;i++){
        result[7-i] = a%2 + '0' - 0;
        a/=2;
    }
    result[8] = '\n';
    return result;
}
int main()
{   
    int fd = open("test.dat",O_CREAT|O_WRONLY);
    if(fd < 0){
        printf("%s\n",strerror(errno));
        exit(-1);
    }
    time_t t; 
    int tempt;
    char buffer[10];
    memset(buffer,0,10);
    int max = -128; 
    int min = 127;
    srand((unsigned) time(&t));
    for(int i=0;i<256;i++){
        tempt = rand();
        tempt = tempt % 256;
        tempt -= 128;
        if(max < tempt)
            max = tempt;
        if(min > tempt)
            min = tempt;
        sprintf(buffer,"%d\n",tempt);
        if(write(fd,buffer,strlen(buffer))<0){
            goto error;
        }
    }
    printf("Max: %d,Min: %d\nMax difference: %d %s\n",max,min,max-min,convert_decimal_binary(max-min));
    close(fd);
    return 0;
error:
    close(fd);
    exit(-1);
}

File: RandomTest.py
import random
import os
# Generate a random integer between -128 and 127

# Specify the file name
file_name = r'E:/Digital_Design__Project/test.dat'
max = -128
min = 127
# Write the random value to the file
with open(file_name, "w+") as file:
    for i in range(0,255):
        random_value = random.randint(-128, 127)
        if max < random_value:
            max = random_value
        if min > random_value:
            min = random_value
        file.write(f"{random_value}\n")
print("Max: "+str(max)+" Min: "+str(min)+" Max diff: "+str(max-min))

